### PR TITLE
Adding UPI protocol support in standard transformer simulator

### PR DESCRIPTION
**Pull request by tiopramayudi**
**What this PR does / why we need it**:

Standard transformer simulator currently only works for HTTP_JSON protocol, this PR is adding support for UPI_V1 protocol also.

**Modification**
* Request conversion to upi PredictValueRequest and update echo mock model predictor for UPI_V1 protocol
* Update ui output of tracing
* Adding button to populate sample request and model predictor payload 
* Adding documentation button
**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

https://user-images.githubusercontent.com/2369255/194526214-2b7c40ec-9c30-44a1-8903-53fed95a3374.mov

```release-note

```

**Checklist**

- [ ] Added unit test, integration, and/or e2e tests
- [x] Tested locally
- [ ] Updated documentation
- [ ] Update Swagger spec if the PR introduce API changes
- [ ] Regenerated Golang and Python client if the PR introduce API changes

### CODE REVIEW

File: charts/merlin/values.yaml [hunk 1]
```diff
@@ -219,6 +219,8 @@ merlin:
       },
     ]
 
+    upiDocURL: "http://sample.doc.com"
```
Comment (pradithya) [inline charts/merlin/values.yaml:222]: Can you make https://github.com/caraml-dev/universal-prediction-interface/blob/main/docs/api_markdown/caraml/upi/v1/index.md as default

File: api/pkg/transformer/executor/transformer.go [hunk 2]
```diff
@@ -80,7 +80,7 @@ func (st *standardTransformer) Execute(ctx context.Context, requestBody types.Pa
 	if env.IsPreprocessOpExist() {
 		preprocessOut, err = env.Preprocess(ctx, requestBody, requestHeaders)
 		if err != nil {
-			return generateErrorResponse(err), err
+			return generateErrorResponse(err), nil
```
Comment (karzuo) [inline api/pkg/transformer/executor/transformer.go:83]: why return nil?

File: api/pkg/transformer/pipeline/encoder_op.go [hunk 3]
```diff
@@ -22,7 +22,7 @@ type Encoder interface {
 func NewEncoderOp(encoders []*spec.Encoder, tracingEnabled bool) *EncoderOp {
 	encoderOp := &EncoderOp{encoderSpecs: encoders}
 	if tracingEnabled {
-		encoderOp.OperationTracing = NewOperationTracing(encoders, types.JsonOutputOpType)
+		encoderOp.OperationTracing = NewOperationTracing(encoders, types.EncoderOpType)
```
Comment (karzuo) [inline api/pkg/transformer/pipeline/encoder_op.go:25]: why is this changed?

File: ui/src/pages/version/components/forms/components/transformer/components/simulation/TransformerSimulationInput.js [hunk 4]
```diff
@@ -11,19 +12,50 @@ import {
...
 
 export const TransformerSimulationInput = ({
   simulationPayload,
+  protocol,
   onChange,
   onSubmit,
   errors
 }) => {
+  const populateSampleRequest = () => {
+    const defaultPayload = {"target_name":"probability","prediction_table":{"name":"prediction_table_sample","columns":[{"name":"col1","type":"TYPE_INTEGER"},{"name":"col2","type":"TYPE_DOUBLE"}],"rows":[{"rowId":"row1","values":[{"integer_value":1},{"doubleValue":4.4}]},{"rowId":"row2","values":[{"integer_value":2},{"doubleValue":8.8}]}]},"transformer_input":{"tables":[{"name":"sample_table","columns":[{"name":"col1","type":"TYPE_INTEGER"},{"name":"col2","type":"TYPE_DOUBLE"}],"rows":[{"rowId":"row1","values":[{"integer_value":1},{"double_value":4.4}]},{"rowId":"row2","values":[{"integer_value":2},{"double_value":8.8}]}]}],"variables":[{"name":"var1","type":"TYPE_INTEGER","integer_value":1},{"name":"var2","type":"TYPE_DOUBLE","double_value":1.1}]},"prediction_context":[{"name":"ctx1","type":"TYPE_INTEGER","integer_value":1},{"name":"ctx2","type":"TYPE_STRING","double_value":"SAMPLE"}]}
```
Comment (karzuo) [inline ui/src/pages/version/components/forms/components/transformer/components/simulation/TransformerSimulationInput.js:26]: so the  generated payload is "fixed"? Are we able to generate one that will actually work based on the yaml?

**Comment by ariefrahmansyah** [comment]
I'm testing your PR on my machine but I cannot see the `+Add UPI Autoload` button. 

image

File: api/pkg/transformer/executor/transformer.go (hunk 2 above)
Comment (tiopramayudi) [reply api/pkg/transformer/executor/transformer.go:83]: i think this one honest mistake, I'll revert this

File: api/pkg/transformer/pipeline/encoder_op.go (hunk 3 above)
Comment (tiopramayudi) [reply api/pkg/transformer/pipeline/encoder_op.go:25]: because it is encoderOp rather than jsonoutputOp

File: ui/src/pages/version/components/forms/components/transformer/components/simulation/TransformerSimulationInput.js (hunk 4 above)
Comment (tiopramayudi) [reply ui/src/pages/version/components/forms/components/transformer/components/simulation/TransformerSimulationInput.js:26]: the aims to give user clue how the valid structure of request will look like, this is something that we can improve in the next iteration

**Comment by tiopramayudi** [comment]
> I'm testing your PR on my machine but I cannot see the `+Add UPI Autoload` button.
> 
> image

Are you using UPI_V1 protocol?

File: api/pkg/transformer/executor/transformer.go (hunk 2 above)
Comment (pradithya) [reply api/pkg/transformer/executor/transformer.go]: Shouldn't it be nil? The simulation response is error but the api response should still be successful right?

Comment (tiopramayudi) [reply api/pkg/transformer/executor/transformer.go]: yeah just realized this, that is trigger alert in ds-critical, I removed the error from the interface completely

**Review by ariefrahmansyah — APPROVED** [review]
LGTM! Thanks for adding protocol selection on transformer tools!
